aten/src/ATen/native/quantized/cpu/qlinear_prepack.cpp

```diff
@@ -157,22 +158,34 @@ class QLinearPackWeightInt8 final : public c10::OperatorKernel {
 
     Tensor weight_contig = weight.contiguous();
     Tensor bias_contig = bias.contiguous();
+    auto weight_zp = weight.q_zero_point() + 128;
 
+    int8_t* inp_data = (int8_t*)weight_contig.data_ptr<c10::qint8>();
+    Tensor qnnp_weight = at::_empty_affine_quantized(
+        weight_contig.sizes(),
+        at::device(kCPU).dtype(kQUInt8),
+        weight.q_scale(),
+        weight_zp);
+    auto* qnnp_w_data = qnnp_weight.data_ptr<c10::quint8>();
+    for (int i = 0; i < weight_contig.numel(); ++i) {
+      qnnp_w_data[i] = static_cast<c10::quint8>(inp_data[i] + 128);
+    }
     initQNNPACK();
 
     auto wt_ptr =
         guts::make_unique<PackedLinearWeightsQnnp>(PackedLinearWeightsQnnp{
             guts::make_unique<qnnpack::PackBMatrix>(
```

Comment:
same comment - it can be nullptr and then you don't even need to quantize bias here. Less code = better :)